TestMessageClientBasicMultiTopics is unstable

### Which jobs are flaking?

TestMessageClientBasicMultiTopics

### Which test(s) are flaking?

leak_test

### Jenkins logs or GitHub Actions link

```log
--- FAIL: TestMessageClientBasicMultiTopics (49.96s)
[2022-04-19T08:32:29.707Z]     server_client_integration_test.go:155: 
[2022-04-19T08:32:29.707Z]         	Error Trace:	server_client_integration_test.go:155
[2022-04-19T08:32:29.707Z]         	            				asm_amd64.s:1571
[2022-04-19T08:32:29.707Z]         	Error:      	Received unexpected error:
[2022-04-19T08:32:29.707Z]         	            	context deadline exceeded
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/errors.AddStack
[2022-04-19T08:32:29.707Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/errors.Trace
[2022-04-19T08:32:29.707Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p/internal.(*SendChan).SendSync
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/internal/send_chan.go:77
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p.(*MessageClient).sendMessage
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/client.go:441
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p.(*MessageClient).SendMessage
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/client.go:385
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p.runP2PIntegrationTest.func6
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/server_client_integration_test.go:154
[2022-04-19T08:32:29.707Z]         	            	runtime.goexit
[2022-04-19T08:32:29.707Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1571
[2022-04-19T08:32:29.707Z]         	Test:       	TestMessageClientBasicMultiTopics
[2022-04-19T08:32:29.707Z]     server_client_integration_test.go:155: 
[2022-04-19T08:32:29.707Z]         	Error Trace:	server_client_integration_test.go:155
[2022-04-19T08:32:29.707Z]         	            				asm_amd64.s:1571
[2022-04-19T08:32:29.707Z]         	Error:      	Received unexpected error:
[2022-04-19T08:32:29.707Z]         	            	context deadline exceeded
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/errors.AddStack
[2022-04-19T08:32:29.707Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/errors.Trace
[2022-04-19T08:32:29.707Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p/internal.(*SendChan).SendSync
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/internal/send_chan.go:77
[2022-04-19T08:32:29.707Z]         	            	github.com/pingcap/tiflow/pkg/p2p.(*MessageClient).sendMessage
[2022-04-19T08:32:29.707Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/client.go:441
[2022-04-19T08:32:29.708Z]         	            	github.com/pingcap/tiflow/pkg/p2p.(*MessageClient).SendMessage
[2022-04-19T08:32:29.708Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/client.go:385
[2022-04-19T08:32:29.708Z]         	            	github.com/pingcap/tiflow/pkg/p2p.runP2PIntegrationTest.func6
[2022-04-19T08:32:29.708Z]         	            		/home/jenkins/agent/workspace/cdc_ghpr_leak_test/go/src/github.com/pingcap/tiflow/pkg/p2p/server_client_integration_test.go:154
```

https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_leak_test/detail/cdc_ghpr_leak_test/24341/pipeline
https://ci.pingcap.net/blue/organizations/jenkins/ut-check/detail/ut-check/9241/pipeline/

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

--- a/pkg/p2p/server_client_integration_test.go
+++ b/pkg/p2p/server_client_integration_test.go
@@ -25,13 +25,12 @@ import (
 	"github.com/phayes/freeport"
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
-	"github.com/stretchr/testify/require"
-	"go.uber.org/zap"
-	"google.golang.org/grpc"
-
 	cerror "github.com/pingcap/tiflow/pkg/errors"
 	"github.com/pingcap/tiflow/pkg/security"
 	"github.com/pingcap/tiflow/proto/p2p"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
 )
 
 // read only

--- a/pkg/p2p/server_client_integration_test.go
+++ b/pkg/p2p/server_client_integration_test.go
@@ -25,12 +25,13 @@ import (
 	"github.com/phayes/freeport"
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
-	cerror "github.com/pingcap/tiflow/pkg/errors"
-	"github.com/pingcap/tiflow/pkg/security"
-	"github.com/pingcap/tiflow/proto/p2p"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
+
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/security"
+	"github.com/pingcap/tiflow/proto/p2p"
 )
 
 // read only
@@ -183,10 +184,10 @@ func TestMessageClientBasic(t *testing.T) {
 }
 
 func TestMessageClientBasicMultiTopics(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.TODO(), defaultTimeout)
+	ctx, cancel := context.WithTimeout(context.TODO(), defaultTimeout*2)
 	defer cancel()
 
-	runP2PIntegrationTest(ctx, t, defaultMessageBatchSizeLarge, 4, 16)
+	runP2PIntegrationTest(ctx, t, defaultMessageBatchSizeLarge, 4, 4)
 }
 
 func TestMessageClientServerRestart(t *testing.T) {

